src/org/jitsi/impl/neomedia/transform/fec/FlexFec03Mask.java

```diff
@@ -0,0 +1,258 @@
+     * method determines the size (in bytes) of the mask in the given packet
+     * buffer and returns it
+     * @param buffer the buffer containing the mask
+     * @param maskOffset the offset in the buffer to the start of the mask
+     * @return the size of the mask, in bytes.
+     */
+    private static int getMaskSizeInBytes(byte[] buffer, int maskOffset)
+        throws MalformedMaskException
+    {
+        if ((buffer.length - maskOffset) < MASK_SIZE_SMALL)
+        {
+            throw new MalformedMaskException();
+        }
+        // The mask is always at least MASK_SIZE_SMALL bytes
+        int maskSizeBytes = MASK_SIZE_SMALL;
+        int kbit0 = (buffer[maskOffset] & 0x80) >> 7;
+        if (kbit0 == 0)
+        {
+            maskSizeBytes = MASK_SIZE_MED;
+            if ((buffer.length - maskOffset) < MASK_SIZE_MED)
+            {
+                throw new MalformedMaskException();
+            }
+            int kbit1 =
+                (buffer[maskOffset + 2] & 0x80) >> 7;
+            if (kbit1 == 0)
+            {
+                if ((buffer.length - maskOffset) < MASK_SIZE_LARGE)
+                {
+                    throw new MalformedMaskException();
+                }
+                maskSizeBytes = MASK_SIZE_LARGE;
+            }
+        }
+        return maskSizeBytes;
+    }
+
+    /**
+     * Determine how big the mask needs to be based on the given base sequence
+     * number and the list of protected sequence numbers
+     * @param baseSeqNum the base sequence number to use for the mask
+     * @param sortedProtectedSeqNums the sequence numbers this mask should mark
+     * as protected. NOTE: this list MUST be in sorted order
+     * @return the size, in bytes, of the mask that is needed to convey
+     * the given protected sequence numbers
+     */
+    private static int getMaskSizeInBytes(int baseSeqNum, List<Integer> sortedProtectedSeqNums)
+    {
+        int largestSeqNum = sortedProtectedSeqNums.get(sortedProtectedSeqNums.size() - 1);
+        int largestDelta = largestSeqNum - baseSeqNum;
+        if (largestDelta <= 14)
+        {
+            return MASK_SIZE_SMALL;
+        }
+        else if (largestDelta <= 45)
+        {
+            return MASK_SIZE_MED;
+        }
+        return MASK_SIZE_LARGE;
+    }
+
+    /**
+     * Get the length of the mask, in bytes
+     * @return the length of the mask, in bytes
+     */
+    public int lengthBytes()
+    {
+        return sizeBytes;
+    }
+
+    /**
+     * Get the list of media packet sequence numbers which are marked
+     * as protected in this mask
+     * @return a list of sequence numbers of the media packets marked as
+     * protected by this mask
+     */
+    public List<Integer> getProtectedSeqNums()
+    {
+        LeftToRightBitSet maskWithoutKBits = getMaskWithoutKBits(maskWithKBits);
+        List<Integer> protectedSeqNums = new ArrayList<>();
+        for (int i = 0; i < this.sizeBytes * 8; ++i)
```

Comment:
Can we loop until sizeBytes*8? The last index (or two, or three) will be invalid due to the k bits. Is maskWithoutKBits.get() just going to return false in this case?

Reply:
it does return false for anything out of range (it only throws if you try and get a negative index).  even still, maybe it would be clearer to specifically use the bit count.

Reply:
fixed this to use `maskWithoutKBits.sizeBits()` since we have it